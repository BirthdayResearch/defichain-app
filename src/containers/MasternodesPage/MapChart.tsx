import React, { useEffect, useState } from "react";
import { csv } from "d3-fetch";
import { scaleLinear } from "d3-scale";
import {
  ComposableMap,
  Geographies,
  Geography,
  Sphere,
  Graticule
} from "react-simple-maps";

const geoUrl =
  "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";

const colorScale = scaleLinear()
  .domain([0.29, 0.68])
  .range(["#F4F3F6", "#5B10FF"]);

const MapChart = () => {
  const [data, setData] = useState([]);

  useEffect(() => {
    csv(`./masternodes.csv`).then(data => {
      setData(data);
    });
  }, []);

  return (
    <ComposableMap>
      <Sphere stroke="#E4E5E6" strokeWidth={0.5} />
      <Graticule stroke="#E4E5E6" strokeWidth={0.5} />
      {data.length > 0 && (
        <Geographies geography={geoUrl}>
          {({ geographies }) =>
            geographies.map(geo => {
              const d = data.find(
                (s: { ISO3: string }) => s.ISO3 === geo.properties.ISO_A3
              );
              return (
                <Geography
                  key={geo.rsmKey}
                  geography={geo}
                  fill={d ? colorScale(d["2017"]) : "#F5F4F6"}
                />
              );
            })
          }
        </Geographies>
      )}
    </ComposableMap>
  );
};

export default MapChart;
