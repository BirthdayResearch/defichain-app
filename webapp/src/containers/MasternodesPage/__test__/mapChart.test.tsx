import React from 'react';
import MapChart from '../components/MapChart';
import { shallow } from 'enzyme';

describe('Map Chart', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<MapChart />);
    expect(wrapper).toMatchSnapshot();
  });
});
