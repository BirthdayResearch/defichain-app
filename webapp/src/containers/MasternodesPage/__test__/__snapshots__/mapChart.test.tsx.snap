// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Chart should check for snapshot 1`] = `
<ComposableMap>
  <Memo(Sphere)
    stroke="#E4E5E6"
    strokeWidth={0.5}
  />
  <Memo(Graticule)
    stroke="#E4E5E6"
    strokeWidth={0.5}
  />
</ComposableMap>
`;
